### ISSUE TITLE
Function "train_model" does not train correctly without "evaluate_during_training = True"

### ISSUE THREAD

**Issue by fritte1993** [ThilinaRajapakse/simpletransformers#675]
Hey and thanks for this great module, its amazingly easy to use!

I am building a regression model from a pretrained Bert encoder and experience the following issue:
If "evaluate_during_training = False", the resulting model seems to stop learning during "train_model" after the first epoch (negative r2 values for both, training and test data, no matter for how many epochs I train the model). Also the running loss is only decreasing during the first epoch.
However, if I enable "evaluate_during_training = True" and pass an eval_df (this can essentially be anything, e.g. part of train or test data, or made up data without an actual meaning), the trained model performs well on both, training and test data after 5 or more epochs.
From my understanding, the "evaluate_during_training" option should not influence the the training of the model (at least not that drastically).

My code looks like the following:
```
settings = {'regression': True, 'overwrite_output_dir': True,
              'train_batch_size': 32, 'eval_batch_size': 32, 'max_seq_length': 512, 'manual_seed': 42,
              'learning_rate': 9.659e-05, 'num_train_epochs': 5, 'evaluate_during_training_verbose': True}
model_args = ClassificationArgs()
model_args.update_from_dict(settings)
model = ClassificationModel('bert', bert_path, num_labels=1, args=model_args)
model.args.evaluate_during_training = True     #needs to be enabled to achieve good performance after training, independed of what is passed as eval_data
model.train_model(train_df=train_data, eval_df=eval_data)
result, model_outputs, wrong_predictions = model.eval_model(test_data, r2=sklearn.metrics.r2_score)
```
Am I missing something completely here, or is this behaviour expected?
Thanks a lot!

**Comment by ThilinaRajapakse** [comment]
It's definitely not expected. You are right in that `evaluate_during_trainng` should have no bearing on the actual training. Does this happen for a particular dataset or is it happening on any dataset?

**Comment by fritte1993** [comment]
Thanks for the quick response! It is happening for all the datasets (and splits) I have tested so far. Interestingly, if I train the model for only one epoch, I get the same r2 values in both cases (evaluate_during_training = True/False). My feeling is that there could be an issue with saving/loading of the models (although I do not set any parameters in this regard).

**Comment by ThilinaRajapakse** [comment]
Training for multiple epochs doesn't cause the model to be loaded from a saved point, so I'm not sure how that could cause this. Looks like this will need to be investigated. Please do update if you find anything else regarding this.
